--- a/src/main.cpp
+++ b/src/main.cpp
@@ -27,6 +27,7 @@
 
 using namespace std;
 using namespace boost;
+using namespace boost::placeholders;
 
 #if defined(NDEBUG)
 # error "Offerings cannot be compiled without assertions."
--- a/src/rpcserver.cpp
+++ b/src/rpcserver.cpp
@@ -28,6 +28,7 @@
 
 using namespace std;
 using namespace boost;
+using namespace boost::placeholders;
 using namespace boost::asio;
 
 static std::string strRPCUserColonPass;
