--- old/qtbase/qmake/library/qmakeevaluator.cpp
+++ new/qtbase/qmake/library/qmakeevaluator.cpp
@@ -1519,7 +1519,7 @@
     feature_roots.removeDuplicates();
 
     QStringList ret;
-    foreach (const QString &root, feature_roots)
+    for (QStringList::const_iterator it = feature_roots.constBegin(); it != feature_roots.constEnd(); ++it)
-        if (IoUtils::exists(root))
-            ret << root;
+        if (IoUtils::exists(*it))
+            ret << *it;
     m_featureRoots = new QMakeFeatureRoots(ret);
 }
