--- old/qtbase/tools/configure/configureapp.cpp
+++ new/qtbase/tools/configure/configureapp.cpp
@@ -2468,8 +2468,11 @@
         success = buildDir.mkdir("config.tests");
 
     QString fileName(buildPath + "/config.tests/.qmake.cache");
-    QFile cacheFile(fileName);
-    success &= cacheFile.open(QIODevice::WriteOnly);
+    QFile cacheFile(fileName);
+    success &= cacheFile.open(QIODevice::WriteOnly | QIODevice::Truncate);
+    if (success) {
+        cacheFile.write("QMAKE_LIBS_QT_ENTRY =\n");
+    }
     cacheFile.close();
 
     if (!success) {
