diff --git a/qtbase/qmake/generators/makefile.cpp b/qtbase/qmake/generators/makefile.cpp
--- a/qtbase/qmake/generators/makefile.cpp
+++ b/qtbase/qmake/generators/makefile.cpp
@@ -1749,7 +1749,7 @@ bool MakefileGenerator::verifyExtraCompiler(const ProString &comp, const QString
         QString tmp_out = project->values(ProKey(comp + ".output")).first().toQString();
         if(tmp_out.isEmpty())
             return false;
-        QString tmp_cmd;
+        QString tmp_cmd = var(ProKey(comp + ".commands"));
         const ProKey ckey(comp + ".commands");
         if (!project->isEmpty(ckey)) {
             int argv0 = -1;
@@ -1762,7 +1762,8 @@ bool MakefileGenerator::verifyExtraCompiler(const ProString &comp, const QString
             }
             if(argv0 != -1) {
                 cmdline[argv0] = Option::fixPathToTargetOS(cmdline.at(argv0).toQString(), false);
-                tmp_cmd = cmdline.join(' ');
+                if (tmp_cmd.isEmpty())
+                    tmp_cmd = cmdline.join(' ');
             }
         }
 
@@ -1832,14 +1833,15 @@ MakefileGenerator::writeExtraCompilerTargets(QTextStream &t)
         QString tmp_out = fileFixify(project->first(ProKey(*it + ".output")).toQString(),
                                      Option::output_dir, Option::output_dir);
         QString tmp_cmd;
-        const ProKey ckey(*it + ".commands");
+        const ProKey ckey(*it + ".commands"); tmp_cmd = var(ckey);
         if (!project->isEmpty(ckey)) {
             QStringList cmdline = project->values(ckey).toQStringList();
             int argv0 = findExecutable(cmdline);
             if(argv0 != -1) {
                 cmdline[argv0] = escapeFilePath(Option::fixPathToTargetOS(cmdline.at(argv0), false));
-                tmp_cmd = cmdline.join(' ');
+                if (tmp_cmd.isEmpty())
+                    tmp_cmd = cmdline.join(' ');
             }
         }
         QString tmp_dep_cmd;
