--- a/qtbase/src/network/socket/qlocalsocket.h
+++ b/qtbase/src/network/socket/qlocalsocket.h
@@ -131,7 +131,12 @@ private:
 #if defined(QT_LOCALSOCKET_TCP)
     Q_PRIVATE_SLOT(d_func(), void _q_stateChanged(QAbstractSocket::SocketState))
     Q_PRIVATE_SLOT(d_func(), void _q_error(QAbstractSocket::SocketError))
-#elif defined(Q_OS_WIN)
+#elif defined(Q_MOC_RUN)
+    /* Qt 5.2.1 moc ignores the Makefile's custom WIN32 defines here. */
+    Q_PRIVATE_SLOT(d_func(), void _q_canWrite())
+    Q_PRIVATE_SLOT(d_func(), void _q_pipeClosed())
+    Q_PRIVATE_SLOT(d_func(), void _q_winError(ulong, const QString &))
+#elif defined(Q_OS_WIN) || defined(_WIN32) || defined(WIN32)
     Q_PRIVATE_SLOT(d_func(), void _q_canWrite())
     Q_PRIVATE_SLOT(d_func(), void _q_pipeClosed())
     Q_PRIVATE_SLOT(d_func(), void _q_winError(ulong, const QString &))
