--- a/qtbase/src/network/ssl/qsslsocket_openssl_p.h
+++ b/qtbase/src/network/ssl/qsslsocket_openssl_p.h
@@ -133,7 +133,7 @@ public:
     QSslCipher sessionCipher() const;
     void continueHandshake();
     bool checkSslErrors();
-#ifdef Q_OS_WIN
+#if defined(Q_OS_WIN) || defined(Q_MOC_RUN) || defined(_WIN32) || defined(WIN32)
     void fetchCaRootForCert(const QSslCertificate &cert);
     void _q_caRootLoaded(QSslCertificate,QSslCertificate);
 #endif
@@ -146,7 +146,7 @@ public:
     static QString getErrorsFromOpenSsl();
 };
 
-#ifdef Q_OS_WIN
+#if defined(Q_OS_WIN) || defined(Q_MOC_RUN) || defined(_WIN32) || defined(WIN32)
 class QWindowsCaRootFetcher : public QObject
 {
     Q_OBJECT;
@@ -160,7 +160,7 @@ private:
     QSslCertificate cert;
     QSslSocket::SslMode mode;
 };
-#endif
+#endif // Q_OS_WIN || Q_MOC_RUN || _WIN32 || WIN32
 
 QT_END_NAMESPACE
 
