--- a/qtbase/qmake/generators/win32/winmakefile.cpp
+++ b/qtbase/qmake/generators/win32/winmakefile.cpp
@@ -283,6 +283,34 @@
     if (!project->values("QMAKE_INCDIR").isEmpty())
         project->values("INCLUDEPATH") += project->values("QMAKE_INCDIR");
 
+    ProStringList qtModules = project->values("QT");
+    qtModules += project->values("QT_PRIVATE");
+    const ProStringList knownQtModules = project->values("QT_MODULES");
+    ProStringList &includePath = project->values("INCLUDEPATH");
+    for (ProStringList::ConstIterator modit = qtModules.constBegin(); modit != qtModules.constEnd(); ++modit) {
+        QString module = (*modit).toQString();
+        if (module.isEmpty())
+            continue;
+        module.replace(QLatin1String("-private"), QLatin1String("_private"));
+
+        QString includeKey = QLatin1String("QT.") + module + QLatin1String(".includes");
+        includePath += project->values(ProKey(includeKey));
+
+        if (module.endsWith(QLatin1String("_private"))) {
+            module.chop(8);
+            QString publicIncludeKey = QLatin1String("QT.") + module + QLatin1String(".includes");
+            includePath += project->values(ProKey(publicIncludeKey));
+        }
+    }
+    for (ProStringList::ConstIterator modit = knownQtModules.constBegin(); modit != knownQtModules.constEnd(); ++modit) {
+        QString module = (*modit).toQString();
+        QString includeKey = QLatin1String("QT.") + module + QLatin1String(".includes");
+        includePath += project->values(ProKey(includeKey));
+        QString privateIncludeKey = QLatin1String("QT.") + module + QLatin1String("_private.includes");
+        includePath += project->values(ProKey(privateIncludeKey));
+    }
+    includePath.removeDuplicates();
+
     if (!project->values("VERSION").isEmpty()) {
         QStringList l = project->first("VERSION").toQString().split('.');
         if (l.size() > 0)
