diff --git a/qtbase/mkspecs/features/qt_app.prf b/qtbase/mkspecs/features/qt_app.prf
--- a/qtbase/mkspecs/features/qt_app.prf
+++ b/qtbase/mkspecs/features/qt_app.prf
@@ -19,6 +19,13 @@ host_build:force_bootstrap {
     contains(QT, core(-private)?|xml) {
         QT -= core core-private xml
         QT += bootstrap-private
+        INCLUDEPATH += \
+            $$MODULE_BASE_OUTDIR/include/QtCore \
+            $$MODULE_BASE_OUTDIR/include/QtCore/$$QT_VERSION \
+            $$MODULE_BASE_OUTDIR/include/QtCore/$$QT_VERSION/QtCore \
+            $$MODULE_BASE_OUTDIR/include/QtXml \
+            $$MODULE_BASE_OUTDIR/include/QtXml/$$QT_VERSION \
+            $$MODULE_BASE_OUTDIR/include/QtXml/$$QT_VERSION/QtXml
     }
     target.path = $$[QT_HOST_BINS]
 } else {
